File: TP1/alteracao/alter.java
import java.util.Random;

class alter {
    @SuppressWarnings("static-access")
    public static void main(String[] args) {
        Random gerador = new Random();
        gerador.setSeed(4);
        MyIO my = new MyIO();
        int tamanho;
        String frase = my.readLine();
        while (!(frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M')) {
            tamanho = frase.length();
            my.println(cifra(frase, tamanho, gerador));
            frase = my.readLine();
        }
    }

    public static String cifra(String frase, int tamanho, Random gerador) {
        String resp = "";
        char orig = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
        char troca = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
        for (int x = 0; x < tamanho; x++) {
            char tempC;
            int tempI;
            if (frase.charAt(x) == orig) {
                tempC = troca;
                tempI = tempC;
                tempC = (char) tempI;
                resp += tempC;
            } else {
                resp += frase.charAt(x);
            }
        }
        return resp;
    }
}

File: TP1/alteracao/alter.c
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

void *alter(char c[], int i, char orig, char troca);

int main()
{

    bool i = true;
    char c[1000];
    srand(4);
    while (i)
    {
        char orig = 'a' + (abs(rand()) % 26);
        char troca = 'a' + (abs(rand()) % 26);
        scanf(" %[^\n]", &c);
        if (c[0] == 'F' && c[1] == 'I' && c[2] == 'M')
        {
            i = false;
        }
        else
        {
            alter(c, 0, orig, troca);
        }
    }
    return 0;
}

void *alter(char c[], int i, char orig, char troca)
{
    int tamanho = strlen(c);
    if (i == tamanho)
    {
        printf("\n");
    }
    else
    {
        if (c[i] == orig)
        {
            c[i] = troca;
        }
        printf("%c", c[i]);
        alter(c, i + 1, orig, troca);
    }
}